#453: Add new data validation: ListFormulaDataValidation.

civgio opened the pull request:
Fixing the [issue 453:](https://github.com/dhatim/fastexcel/issues/453): a new data validation is added: `ListFormulaDataValidation`.
This `DataValidation` allow to create a data validation with `list` type and with a `formula` value (to referrer a list, for example with `INDIRECT` formula).

In addition, it is possibility to set global visibility to a range when a named range is added.

civgio commented:
@meiMingle @rzymek can you check the pull request and if you think the change makes sense, merge it into master?

rzymek commented on fastexcel-writer/src/main/java/org/dhatim/fastexcel/ListFormulaDataValidation.java:
```diff
@@ -0,0 +1,127 @@
+    public void write(Writer w) throws IOException {
+        w
+                .append("<dataValidation sqref=\"")
+                .append(range.toString())
```
unnessesary .toString()

civgio replied:
If I try to append the range without "toString", an error is return

rzymek commented on fastexcel-writer/src/main/java/org/dhatim/fastexcel/Range.java:
```diff
@@ -216,13 +233,32 @@ public CustomDataValidation validateWithFormula(String formula) {
+     */
+    public Range setGlobalVisibility(boolean globalVisibility) {
+        this.globalVisibility = globalVisibility;
+        return this;
```
other setters return void. returning this here seems inconsistent

civgio replied:
I understand. I change it

rzymek commented on fastexcel-writer/src/main/java/org/dhatim/fastexcel/Range.java:
```diff
@@ -44,6 +44,11 @@ public class Range implements Ref {
     private final int right;
 
+    /**
+     * Check about the Range visibility when a named range is added.
```
I find this comment confusing. Doesn't seem to turn on any "Check", but rather control the scope of this range (right?)

rzymek reviewed (commented):
I would be nice if you'd add some tests covering those changes

civgio commented:
@rzymek thank you for your comments and your time.
I have fixed all your issues and I have added the tests.
I hope all is ok. If you have any other doubts, I am here.
Have a nice evening.

civgio commented:
Please @rzymek, can you close the pull request if all is ok? Otherwise can you tell me what needs to be fix to complete the pull request?
Thank you so much for your support and availability
